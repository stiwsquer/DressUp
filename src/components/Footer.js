import React from "react";

export default function Footer(props) {
  return (
    <footer>
      <div className="footer-wrapper">
        <h5>DressUp &copy;</h5>
        <ul>
          <li>
            <a href="#" title="twitter-social-media">
              {/* <img src="../icons/twitter.svg" alt="twitter-social-media" /> */}
            </a>
          </li>
          <li>
            <a href="#" title="youtube-social-media">
              {/* <img src="../icons/youtube.svg" alt="youtube-social-media" /> */}
            </a>
          </li>
          <li>
            <a href="#" title="instagram-social-media">
              <img
                src={require("../icons/heart.png")}
                alt="instagram-social-media"
              />
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
}
