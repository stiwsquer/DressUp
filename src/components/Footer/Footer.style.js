import styled from "styled-components";
import { flex } from "../../GlobalStyles";

export const FooterSection = styled.footer`
  background: #202020;

  color: white;
`;

export const FooterWrapper = styled.div`
  min-height: 15vh;
  width: 95%;
  margin: auto;
  font-size: 2.5rem;

  ${flex({ justify: "space-between", align: "center" })}
  flex-wrap: wrap;

  h1 {
    flex: 2 1 40rem;
    margin: 2rem 0rem;
  }

  ul {
    flex: 1 2 40rem;
    ${flex({ justify: "space-around", align: "center" })}
    list-style: none;
  }

  a {
    color: white;
  }

  .social-media {
    color: white;
    font-size: 3rem;
    opacity: 0.5;
    margin: 2rem;
    &:hover {
      color: #f8bb99;
    }
  }

  @media screen and (max-width: 850px) {
    h1 {
      text-align: center;
    }
  }
`;
