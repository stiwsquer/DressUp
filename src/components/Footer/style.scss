@import "../../styles/global.scss";
footer {
  background: #202020;

  color: white;
}
.social-media {
  color: white;
  font-size: 3rem;
  opacity: 0.5;
  margin: 2rem;
  &:hover {
    color: #f8bb99;
  }
}

.footer-wrapper {
  min-height: 15vh;
  width: 95%;
  margin: auto;
  @include flex(row, space-between, center);
  flex-wrap: wrap;

  h1 {
    flex: 2 1 40rem;
    margin: 2rem 0rem;
  }

  ul {
    flex: 1 2 40rem;
    @include flex(row, space-around, center);
    list-style: none;
  }

  a {
    color: white;
  }
}

@media screen and (max-width: 850px) {
  .footer-wrapper {
    h1 {
      text-align: center;
    }
  }
}
