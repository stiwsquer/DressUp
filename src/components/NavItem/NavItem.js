import React, { useContext } from "react";
import { NavLink, Link } from "react-router-dom";
export default React.memo(function NavItem({ linkTo, iconClass, text }) {
  return (
    <Link to={linkTo}>
      <i className={iconClass}></i>
      {iconClass && " "}
      {text}
    </Link>
  );
});
