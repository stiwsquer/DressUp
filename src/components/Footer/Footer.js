import React from "react";
import NavItem from "../NavItem/NavItem";
import "./style.scss";

export default function Footer() {
  return (
    <footer>
      <div className="footer-wrapper">
        <h1>
          <NavItem text="DressUp" linkTo="#" /> &copy;
        </h1>
        <ul>
          <li>
            <NavItem iconClass="fab fa-twitter social-media" linkTo="#" />
          </li>
          <li>
            <NavItem iconClass="fab fa-instagram social-media" linkTo="#" />
          </li>
          <li>
            <NavItem iconClass="fab fa-youtube social-media" linkTo="#" />
          </li>
        </ul>
      </div>
    </footer>
  );
}
