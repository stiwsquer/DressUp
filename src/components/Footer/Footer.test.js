import Footer from "./Footer";
import NavItem from "../NavItem/NavItem";

it("renders Footer component without crashing", () => {
  shallow(<Footer />);
});

it("renders four <NavItem /> components", () => {
  const wrapper = shallow(<Footer linkTo="#" />);
  expect(wrapper.find(NavItem)).toHaveLength(4);
});

it("contains footer element", () => {
  const wrapper = shallow(<Footer linkTo="#" />);
  expect(wrapper.find("footer")).toHaveLength(1);
});

it("contains footer element", () => {
  const wrapper = shallow(<Footer linkTo="#" />);
  expect(wrapper.find("li")).toHaveLength(3);
});

it("Footer snapshot - shallow", () => {
  const wrapper = shallow(<Footer />);
  expect(wrapper).toMatchSnapshot();
});
