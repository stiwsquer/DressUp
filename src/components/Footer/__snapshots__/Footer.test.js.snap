// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot - shallow 1`] = `
<styled.footer>
  <styled.div>
    <h1>
      <Memo(NavItem)
        linkTo="/"
        text="DressUp"
      />
       Â©
    </h1>
    <ul>
      <li>
        <Memo(NavItem)
          iconClass="fab fa-twitter social-media"
          linkTo="#"
        />
      </li>
      <li>
        <Memo(NavItem)
          iconClass="fab fa-instagram social-media"
          linkTo="#"
        />
      </li>
      <li>
        <Memo(NavItem)
          iconClass="fab fa-youtube social-media"
          linkTo="#"
        />
      </li>
    </ul>
  </styled.div>
</styled.footer>
`;
